src/extensions/as-legacy-globals-for-extension-api/legacy-global-di-for-extension-api.ts

```diff
@@ -20,13 +21,17 @@ export const setLegacyGlobalDiForExtensionApi = (
 };
 
 export const getLegacyGlobalDiForExtensionApi = () => {
-  const globalDis = [...legacyGlobalDis.values()];
-
-  if (globalDis.length > 1) {
+  if (legacyGlobalDis.size > 1) {
     throw new Error("Tried to get DI container using legacy globals where there is multiple containers available.");
   }
 
-  return globalDis[0];
+  const di = iter.first(legacyGlobalDis.values());
```

Comment:
Iter is wrong tool for the job, causing complexity, and code that is not left-to-right, top-to-bottom -readable.

Reply:
Disagree, this is no different than your pipeline in terms of complexity or direction of code readability.

Reply:
1. It's not "our pipeline", it's a classical functional pattern.
2. `iter` is the wrong tool, because complexity of generators does not solve anything here.

How about the proposal to experiment with generators somewhere else? :)

Reply:
Stop saying "generators", it is a standard iterator adapter pattern. I disagree that it doesn't solve anything here as the function is quite readable in what it tries to accomplish.

Just because it is a functional pattern doesn't mean that it doesn't have its own complexities that declarative programming doesn't solve either.